### PR TITLE
Add .NET 6 to target frameworks for rules and recommendations

### PR DESCRIPTION
**Pull request by jonlouie**
**Issue #, if available:**
Closes: #121 

**Description of changes:**
* Rules and recommendations have .NET 6 added as a target framework if TargetFrameworks already includes .NET 5.
* Renamed validator objects from *POJO.cs to *POCO.cs to match C# convention
* Added a script to add TargetFrameworks to rules and recommendation files

**Note:**
* Will be easier to digest this PR by viewing each commit rather than looking at all files in a unified pass.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Eruanion — DISMISSED** [review]
Going by commit was definitely a very useful approach, thanks Jon.

**Review by Eruanion — APPROVED** [review]

File: RecommendationTools/RecommendationBatchUpdater/Updater.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                                    Name = frameworkToAdd,
+                                    TargetCPU = Default.TargetCPU
+                                });
+                            }
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"Failed to save rule to {filePath}: {e}");
```
Comment (saikiranakula-amzn) [inline RecommendationTools/RecommendationBatchUpdater/Updater.cs:62]: comment should be: "Failed to Add framework to {filePath}"

File: RecommendationTools/RecommendationBatchUpdater/Program.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Parser.Default.ParseArguments<Options>(args)
+                .WithParsed<Options>(o =>
+                {
+                    try
+                    {
+                        // ADD/MODIFY CODE HERE TO UPDATE RECOMMENDATIONS AND/OR RULES FILES
```
Comment (saikiranakula-amzn) [inline RecommendationTools/RecommendationBatchUpdater/Program.cs:15]: Add a comment here: "We need to check the package compatibility before Updating the target framework on all the rules files"

**Review by saikiranakula-amzn — CHANGES_REQUESTED** [review]

File: RecommendationTools/RecommendationBatchUpdater/Program.cs (hunk 2 above)
Comment (jonlouie) [reply RecommendationTools/RecommendationBatchUpdater/Program.cs]: Good catch- fixed

File: RecommendationTools/RecommendationBatchUpdater/Updater.cs (hunk 1 above)
Comment (jonlouie) [reply RecommendationTools/RecommendationBatchUpdater/Updater.cs]: Good catch- fixed

**Review by saikiranakula-amzn — APPROVED** [review]

**Pull request merged**